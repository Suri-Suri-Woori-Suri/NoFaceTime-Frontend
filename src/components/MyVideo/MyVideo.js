import React from 'react';
import styles from './MyVideo.module.css';
import { BACKGROUND_IMG } from '../../constants';

const MyVideo = ({
  isHost,
  videoRef,
  canvasRef,
  isMuted
}) => {
  return (
    <div className={styles.Wrapper}>
      <video
        className={isHost ? styles.Host : styles.Peer}
        ref={videoRef}
        muted={isMuted}
        width={500}
        height={375}
        poster={BACKGROUND_IMG.LOADING}
        autoPlay
        playsInline
        data-testid='video'
      />
      <canvas
        ref={canvasRef}
        className={styles.Canvas}
        width={500}
        height={375} />
    </div>
  );
};

export default MyVideo;
