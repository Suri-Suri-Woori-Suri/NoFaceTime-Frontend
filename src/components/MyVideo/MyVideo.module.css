.PeerWrapper {
  position: relative;
}

.Host {
  background-color: black;
  width: 100%;
  height: 100%;
  transform: rotateY(180deg);
  -webkit-transform: rotateY(180deg);
  -moz-transform: rotateY(180deg);
}

.Peer {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border: 10px solid green;
  box-sizing: border-box;
  transform: rotateY(180deg);
  -webkit-transform: rotateY(180deg);
  -moz-transform: rotateY(180deg);
}

canvas {
  transform: rotateY(180deg);
  -webkit-transform: rotateY(180deg);
  -moz-transform: rotateY(180deg);
}
