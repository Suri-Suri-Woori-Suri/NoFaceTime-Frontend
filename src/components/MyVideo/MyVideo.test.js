import React from 'react';
import { mount } from 'enzyme';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import MyVideo from './MyVideo';

describe('<MyVideo />', () => {
  afterEach(cleanup);

  const mockHandleVideoPlay = jest.fn();

  const wrapper = mount(
    <MyVideo
      isHost={true}
      onPlay={mockHandleVideoPlay}
      autioMuted='muted'
    />
  );

  it('should be invoked when video is played', () => {
    render(
      <MyVideo
        isHost={true}
        handleVideoPlay={mockHandleVideoPlay}
        audioMuted='muted'
      />
    );

    fireEvent.play(screen.getByTestId('video'));

    expect(mockHandleVideoPlay).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
