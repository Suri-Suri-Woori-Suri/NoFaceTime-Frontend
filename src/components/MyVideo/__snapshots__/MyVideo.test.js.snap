// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyVideo /> matches snapshot 1`] = `
<MyVideo
  autioMuted="muted"
  isHost={true}
  onPlay={[MockFunction]}
>
  <div
    className="Wrapper"
  >
    <video
      autoPlay={true}
      className="Host"
      data-testid="video"
      height={375}
      playsInline={true}
      poster="https://two-face-time.s3.ap-northeast-2.amazonaws.com/background-image/loading.png"
      width={500}
    />
    <canvas
      className="Canvas"
      height={375}
      width={500}
    />
  </div>
</MyVideo>
`;
